tests/utilities/test_fetching.py

```diff
@@ -251,7 +251,7 @@ def training_step(self, dataloader_iter, batch_idx):
 
         def training_epoch_end(self, *_):
             assert self.trainer.fit_loop.epoch_loop.batch_progress.current.ready == 33
-            assert self.trainer._data_connector.train_data_fetcher.fetched == 64
+            assert self.trainer.fit_loop._data_fetcher.fetched == 64
```

Comment:
do we need similar tests for eval?

Reply:
controlling the dataloader in the `step` is only supported for training

Reply:
oh! correct, I didn't see that it uses just the `DataFetcher`